File: 1365. How Many Numbers Are Smaller Than the Current Number.kt
class Solution {
    fun smallerNumbersThanCurrent(nums: IntArray): IntArray {
        var result = IntArray(nums.size)
        
        for (i in nums.indices){
            var count=0
            for (j in nums.indices){
                if (i!=j&&nums[i]>nums[j]){
                    count++;
                }
            }
            result[i]=count
        }
        return result
    }
}

File: 1365. How Many Numbers Are Smaller Than the Current Number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        sorted_nums = sorted(nums)
        num_map = {}
        for i,j in enumerate(sorted_nums):
            if j not in num_map:
                num_map[j]=i
        result=[num_map[num] for num in nums]

        return result
